Add `teocns/neocursor.nvim` (#2432)
rockerBOO/awesome-neovim: README.md

## AI

- [cursortab/cursortab.nvim](https://github.com/cursortab/cursortab.nvim) - Edit completions and cursor predictions with multiple AI providers.

> - [teocns/neocursor.nvim](https://github.com/teocns/neocursor.nvim) - Next-edit predictions, cursor jumps, and ghost text from Cursor's Tab model, driven by an existing Cursor session instead of an API key.

Reviewer: ```suggestion
- [teocns/neocursor.nvim](https://github.com/teocns/neocursor.nvim) - Next-edit predictions, cursor jumps, and ghost text, all driven by an existing Cursor session instead of an API key.
```

Author: Applied verbatim in ac8724c — thanks, it reads better without "Cursor" twice in one line.
